crates/task-impls/src/consensus.rs

```diff
@@ -1380,6 +1367,16 @@ where
         }
 
         let parent_leaf = leaf.clone();
+        let parent_header = match parent_leaf.deltas {
```

Comment:
I think we should refactor it so `parent_leaf.deltas` *always* contains the header and optionally the payload as well.

Actually maybe just replace `Leaf.deltas` with `Leaf.header`, and the add an `Option<Payload>` on  the Decide event? I don't think we ever actually need the payload in the leaf data structure itself, it's just like an additional thing that may or may not be published along with the leaf when we reach decide. But we should always have the header so we should be able to enforce that in the type system and eliminate this runtime failure mode

Reply:
`VIDBlockPayload` contains the list of transactions and the commitment, the latter of which is already used to create the header, so there's a redundancy if we are going to always include the header in the leaf. I think it may be better to make the leaf have a header and optional transactions, rather than an optional payload.

Reply:
Oh, at a quick search of the use of `deltas`, transaction commitments should suffice, so the leaf could just have `header` and `Option<VIDTransaction>`.

Reply:
I guess if the full thing isn't needed by consensus, it can also go in the Decide event but not the leaf. But we do need some way for the application to get the full payload, in order (ie not `HashSet`). Ideally for simplicity we would just use the same data structure all over (`VIDBlockPayload`). We especially don't have to worry too much about a little redundancy if we only put in the Decide event, since that is just for communication and doesn't get persisted anywhere.

Reply:
Update: the leaf now has a header and optional payload.